src/main/java/com/kobylynskyi/graphql/codegen/mapper/InputDefinitionToDataModelMapper.java

```diff
@@ -56,8 +63,20 @@ public InputDefinitionToDataModelMapper(MapperFactory mapperFactory,
         dataModel.put(CLASS_NAME, dataModelMapper.getModelClassNameWithPrefixAndSuffix(mappingContext, definition));
         dataModel.put(JAVA_DOC, JavaDocBuilder.build(definition));
         dataModel.put(NAME, definition.getName());
-        dataModel.put(FIELDS, inputValueDefinitionToParameterMapper
-                .map(mappingContext, definition.getValueDefinitions(), definition.getName()));
+        List<ParameterDefinition> fields = inputValueDefinitionToParameterMapper
+                .map(mappingContext, definition.getValueDefinitions(), definition.getName());
+        if (mappingContext.isSupportUnknownFields()){
+            ParameterDefinition unknownFields = new ParameterDefinition();
+            unknownFields.setName(mappingContext.getUnknownFieldsPropertyName());
+            unknownFields.setOriginalName(mappingContext.getUnknownFieldsPropertyName());
+            unknownFields.setType("java.util.Map<String,Object>");
+            unknownFields.setAnnotations(Arrays.asList(
+                    JsonAnySetter.class.getName(),
+                    JsonAnyGetter.class.getName()
```

Comment:
Shouldn't this also be configurable?

Reply:
I am not sure what you mean for configurable. Both annotations are required to handle unknown properties, and they only take a `boolean enabled() default true;` argument. What type of configuration are you referring to?

Reply:
I had a look at Jackson to see if there are other ways of configuring support for unknown properties, and I think the two annotations above is the only option. I'll resolve this conversation on the assumption that the code is ok as is, but happy to change it if needed.

Reply:
@aldib I was taking about ability for users to choose which annotation to use. What if someone prefers other ser/deser provider like Gson?

Reply:
Hi,

I agree with your idea, but the dependency on Jackson is already hard coded in a number of places. For example:
* https://github.com/kobylynskyi/graphql-java-codegen/blob/97fef9681501a883e49ab4a482f4e5d01e982230/src/main/java/com/kobylynskyi/graphql/codegen/parser/GraphQLDocumentParser.java#L3
* https://github.com/kobylynskyi/graphql-java-codegen/blob/97fef9681501a883e49ab4a482f4e5d01e982230/src/main/java/com/kobylynskyi/graphql/codegen/supplier/MergeableMappingConfigSupplier.java#L3

If you see it as a requirement, I can add an additional flag to specify the serialization library (e.g. jackson vs gson), and modify the factory method accordingly. However, I think that it might introduce inconsistency unless the same principle is applied everywhere else.

If you are happy with the code as is, I would appreciate if you could resolve this conversation and authorize the next steps of the merge process.

Thank you.

Reply:
@kobylynskyi  the build is failing because of `No sonar.login or SONAR_TOKEN env variable was set`. Can you please advise on how to proceed?